grades/api.py

```diff
@@ -202,3 +204,33 @@ def freeze_user_final_grade(user, course_run, raise_on_exception=False):
         course_run_paid_on_edx=final_grade.payed_on_edx
     )
     return final_grade_obj
+
+
+def generate_program_certificate(user, program):
+    """
+    Create a program certificate if the user has a MM course certificate
+    for each course in the program
+
+    Args:
+        user (User): a Django user.
+        program (programs.models.Program): program where the user is enrolled.
+    """
+    mmtrack = get_mmtrack(user, program)
+
+    if MicromastersProgramCertificate.objects.filter(user=user, program=program).exists():
+        log.error('User [%s] already has a certificate for program [%s]', user, program)
+        return
+
+    for course in program.course_set.all():
+        final_grades = mmtrack.get_passing_final_grades_for_course(course)
+        if not final_grades.exists():
```

Comment:
This pattern isn't ideal since `final_grades.first()` could be `None` if something changed in between this statement and the next. Instead can you check if `final_grades.first()` is `None`?